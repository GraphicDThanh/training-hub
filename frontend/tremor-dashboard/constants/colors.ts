type ColorChannelsChartType = {
  [key: number]: string;
  1: string;
  2: string;
  3: string;
  4: string;
};

export const COLOR_CHANNELS_CHART: ColorChannelsChartType = {
  1: "blue-500",
  2: "red-500",
  3: "primary",
  4: "gray-500",
};
