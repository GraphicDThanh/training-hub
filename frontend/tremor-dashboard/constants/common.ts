// Types
import { OptionType } from "@/types";

export const EXCEPT_KEYS = {
  POSITIVE_INTEGER: ["e", "E", "+", "-", "."],
  POSITIVE_DOUBLE: ["e", "E", "+", "-"],
};

export enum DIRECTION {
  ASC = "asc",
  DESC = "desc",
}

// Mailto Support
export const MAILTO_SUPPORT = "support@tremor-dashboard.com";

export const REMEMBER_ME_COOKIES_KEY = "remember-me";
export const UID_KEY = "uid";
export const ROLE = "role";
export const NAME = "name";
export const AVATAR = "avatar";
export const AUTH_SESSION_COOKIES_KEY = "authjs.session-token";

export const ORDER_LIST_OPTIONS: OptionType[] = [
  { option: "Paid", value: "0" },
  { option: "Refunded", value: "2" },
  { option: "Canceled", value: "1" },
];

export const PRODUCT_LIST_OPTIONS: OptionType[] = [
  { option: "Yes", value: "true" },
  { option: "No", value: "false" },
];

export const FORMAT_DATE = {
  YEAR_MONTH_DATE_TIME: "YYYY-MM-DDTHH:mm:ss.SSS",
  MONTH_DAY_YEAR_TIME: "MMM D, YYYY h:mm A",
};

export const SEPARATOR = {
  COMMAS: ",",
  DOTS: ".",
};

export const CDN_KEY = process.env.NEXT_PUBLIC_CDN_KEY;

export const DRAG_ZONE = {
  DEFAULT: {
    TEXT: "Drag and drop your file here to begin uploading.",
    STYLE_INPUT: "border-gray-300",
    STYLE_TEXT: "text-gray-500 dark:text-gray-400",
  },
  ON_DRAG: {
    TEXT: "Release the mouse click to upload the photo",
    STYLE_INPUT: "outline outline-2 outline-[#ebeff4]",
    STYLE_TEXT: "text-gray-500 dark:text-gray-400",
  },
  ON_WRONG_FORMAT: {
    TEXT: "Invalid image file type. Please try uploading again.",
    STYLE_INPUT: "border-red-500",
    STYLE_TEXT: "text-red-500 dark:text-red-500",
  },
};

export const TRANSACTION_CLASS = {
  INCREASE: {
    buttonClasses: "text-few dark:text-few dark:hove:text-few border-few",
    textClasses: "font-semibold text-greener dark:text-few",
  },
  DECREASE: {
    buttonClasses:
      "text-attention dark:text-attention dark:hover:text-attention border-attention",
    textClasses: "font-semibold text-blood dark:text-pink",
  },
  PENDING: {
    buttonClasses:
      "text-primary dark:text-primary dark:hover:text-primary border-primary",
    textClasses:
      "font-semibold text-primary dark:text-dark-tremor-content-pending",
  },
};

export const enum VARIANT_BUTTON {
  LIGHT = "light",
  LIGHT_CARD = "lightCard",
  LIGHT_CENTER = "lightCenter",
  LIGHT_STATUS = "lightStatus",
  SECONDARY = "secondary",
  SECONDARY_SHADOW = "secondaryShadow",
  PRIMARY = "primary",
  PRIMARY_CENTER = "primaryCenter",
  SURFACE = "surface",
  DARK = "dark",
  ACTIVE = "active",
}

export const NO_IMAGE = "No Image";

export const RESULT_NOT_FOUND = "Result Not Found";

export const AMOUNT_LIMIT = 20000;
export const PRICE_LIMIT = 20000;
export const QUANTITY_LIMIT = 10000;
export const WEIGHT_LIMIT = 2000;

export const PREFIX_PHONE_NUMBER = "+1";

export const STORAGES = {
  IS_DARK_THEME: "isDarkTheme",
};

export const THEMES = {
  LIGHT: "light",
  DARK: "dark",
};

export const LOGO_TITLE = "Tremor Dashboard PRO";
