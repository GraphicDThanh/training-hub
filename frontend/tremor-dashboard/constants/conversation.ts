export const CONVERSATION_TABLES = {
  GROUPS: "groups",
  CHATS: "chats",
};

export const CONVERSATION_ORDERS = {
  CREATED_AT: "createdAt",
  UPDATED_AT: "updatedAt",
};

export const CONVERSATION_QUERYS = {
  USER_IDS: "userIds",
};

export enum CONVERSATION_STATUS {
  INITIAL = "initial",
  FETCHED = "fetched",
}

export const ADMIN_ID = parseInt(process.env.NEXT_PUBLIC_ADMIN_ID!);

export enum DOCUMENT_CHANGE_TYPE {
  ADDED = "added",
  MODIFIED = "modified",
  REMOVED = "removed",
}

export enum WHERE_FILTER_OP {
  ARRAY_CONTAINS = "array-contains",
}
