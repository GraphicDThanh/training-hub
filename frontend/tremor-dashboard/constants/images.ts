export const NOT_FOUND_IMAGE = "/images/no-image-square.webp";
export const LOGO_IMAGE = "/assets/images/sidebar-logo.webp";
export const PLACEHOLDER_IMAGE = "/images/image-regular.svg";

export const AVATAR_IMAGE = {
  XS: "/images/avatar/avatar-xs.webp",
  SM: "/images/avatar/avatar-sm.webp",
  MD: "/images/avatar/avatar-md.webp",
  LG: "/images/avatar/avatar-lg.webp",
};

export const METADATA_IMAGE = {
  HOME_PREVIEW: "/assets/images/home-preview.webp",
  MASTERCARD: "/assets/images/mastercard.webp",
};

export const PROJECT_IMAGE = "/assets/images/profile-project.webp";
