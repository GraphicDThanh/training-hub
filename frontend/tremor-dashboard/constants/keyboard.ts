export const KEYBOARD_KEYS = {
  BACKSPACE: "Backspace",
  DIGIT_1: "1",
};
