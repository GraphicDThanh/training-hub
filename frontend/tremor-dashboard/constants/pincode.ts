export const PIN_CODE_DEFAULT = "*";
export const PIN_CODE_LENGTH = 6;
export const PIN_CODE_SKIP_SET_KEY = "skipSetupPinCode";
