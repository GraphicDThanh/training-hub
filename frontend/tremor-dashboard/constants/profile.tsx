import { PlatformSetting } from "@/types";

export const SETTING_FIELDS_ACCOUNT: PlatformSetting[] = [
  { label: "Email me when someone follows me", field: "emailMentions" },
  {
    label: "Email me when someone answers on my post",
    field: "emailFollowing",
  },
  { label: "Email me when someone mentions me", field: "emailAnswerPost" },
];

export const SETTING_FIELDS_APPLICATION: PlatformSetting[] = [
  { label: "New launches and projects", field: "newLaunchesProject" },
  { label: "Monthly product updates", field: "monthlyProductUpdate" },
  { label: "Subscribe to newsletter", field: "subscribeToNewsletter" },
];

export const SETTING_DATA_ACCOUNT = {
  emailMentions: false,
  emailFollowing: false,
  emailAnswerPost: false,
};

export const SETTING_DATA_APPLICATION = {
  newLaunchesProject: false,
  monthlyProductUpdate: false,
  subscribeToNewsletter: false,
};

// Social links
export const SOCIAL_LINKS = {
  facebook: "https://www.facebook.com/",
  twitter: "https://twitter.com/",
  instagram: "https://instagram.com",
};
