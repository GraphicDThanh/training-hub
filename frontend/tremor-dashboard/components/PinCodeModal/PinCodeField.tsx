"use client";

import { KeyboardEvent, forwardRef, memo, useState } from "react";

import { PIN_CODE_DEFAULT, KEYBOARD_KEYS } from "@/constants";

import "./style.css";
import { isNumber } from "@/helpers";

export interface IPinCodeField {
  index: number;
  value?: string;
  onChange: (value: string, index: number) => void;
  autoFocus?: boolean;
  onFocus?: () => void;
}

export const PinCodeField = memo(
  forwardRef<HTMLInputElement, IPinCodeField>(
    ({ index, value = "", autoFocus = false, onFocus, onChange }, ref) => {
      const [currentValue, setCurrentValue] = useState(value);
      const [isFocus, setIsFocus] = useState(autoFocus);

      const handleFocus = () => {
        onFocus?.();
        setIsFocus(true);
      };

      const handleBlur = () => {
        setIsFocus(false);
      };

      const handleKeyDown = (e: KeyboardEvent<HTMLInputElement>) => {
        const { key } = e;

        const { BACKSPACE } = KEYBOARD_KEYS;

        if (key === BACKSPACE) {
          onChange(PIN_CODE_DEFAULT, index);
          setCurrentValue(PIN_CODE_DEFAULT);
          return;
        }

        if (key !== "e" && isNumber(key)) {
          onChange(key, index);
          setCurrentValue(key);
          return;
        }

        e.preventDefault();
      };

      return (
        <input
          data-testid="pin-code-input"
          ref={ref}
          value={isFocus || value === PIN_CODE_DEFAULT ? "" : currentValue}
          onChange={() => {}}
          autoFocus={autoFocus}
          type={"password"}
          inputMode="numeric"
          placeholder="â—‹"
          maxLength={1}
          className={`w-8 sm:w-full h-8 sm:h-10 border-2 rounded-md outline-none border-secondary focus:border-tremor-secondary text-center font-bold text-2xl text-primary dark:text-white placeholder-secondary focus:placeholder-transparent dark:bg-transparent dark:border-white focus:dark:border-tremor-secondary dark:placeholder-white focus:dark:placeholder-transparent${
            isFocus ? " focused" : ""
          }`}
          onFocus={handleFocus}
          onKeyDown={handleKeyDown}
          onBlur={handleBlur}
        />
      );
    },
  ),
);
