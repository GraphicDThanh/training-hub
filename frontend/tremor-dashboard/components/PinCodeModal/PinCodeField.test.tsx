import React from "react";
import { render, fireEvent, cleanup } from "@testing-library/react";
import { IPinCodeField, PinCodeField } from "./PinCodeField";
import { PIN_CODE_DEFAULT, KEYBOARD_KEYS } from "@/constants";

const { BACKSPACE, DIGIT_1 } = KEYBOARD_KEYS;

let onChange: () => void;

beforeEach(() => {
  cleanup();
  onChange = jest.fn();
});

const setup = (overrideProps?: Partial<IPinCodeField>) => {
  return render(
    <PinCodeField index={0} onChange={onChange} {...overrideProps} />,
  );
};

describe("PinCodeField", () => {
  it("Should match snapshot", () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });

  it("Handles change event correctly", () => {
    const { getByTestId } = setup();
    const inputElement = getByTestId("pin-code-input");

    fireEvent.keyDown(inputElement, { key: DIGIT_1 });

    expect(onChange).toHaveBeenCalledWith(DIGIT_1, 0);
  });

  it("Handles backspace key correctly", () => {
    const { getByTestId } = setup();
    const inputElement = getByTestId("pin-code-input");

    fireEvent.keyDown(inputElement, { key: BACKSPACE, code: BACKSPACE });

    expect(onChange).toHaveBeenCalledWith(PIN_CODE_DEFAULT, 0);
  });

  it("Should clear input on focus", () => {
    const { getByTestId } = setup({ value: DIGIT_1 });
    const inputElement = getByTestId("pin-code-input") as HTMLInputElement;
    fireEvent.focus(inputElement);
    expect(inputElement.value).toBe("");
  });

  it("Should restores input value on blur", () => {
    const { getByTestId } = setup({ value: DIGIT_1 });
    const inputElement = getByTestId("pin-code-input") as HTMLInputElement;
    fireEvent.blur(inputElement);
    expect(inputElement.value).toBe(DIGIT_1);
  });

  it("Should switch input type to password if value is not default", () => {
    const { getByTestId } = setup({ value: DIGIT_1 });
    const inputElement = getByTestId("pin-code-input");
    expect(inputElement.getAttribute("type")).toBe("password");
  });

  it('onChange triggered when keydown is number or "Backspace"', () => {
    const { getByTestId } = setup();
    const inputElement = getByTestId("pin-code-input");

    fireEvent.keyDown(inputElement, { key: DIGIT_1 });
    expect(onChange).toHaveBeenCalledTimes(1);

    fireEvent.keyDown(inputElement, { key: BACKSPACE });
    expect(onChange).toHaveBeenCalledTimes(2);
  });

  it('OnChange not triggered when keydown not is number or "Backspace"', () => {
    const { getByTestId } = setup();
    const inputElement = getByTestId("pin-code-input");

    fireEvent.keyDown(inputElement, { key: "e" });

    expect(onChange).not.toHaveBeenCalled();
  });
});
