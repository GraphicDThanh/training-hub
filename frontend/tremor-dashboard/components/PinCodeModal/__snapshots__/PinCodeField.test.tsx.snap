// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PinCodeField Should match snapshot 1`] = `
<div>
  <input
    class="w-8 sm:w-full h-8 sm:h-10 border-2 rounded-md outline-none border-secondary focus:border-tremor-secondary text-center font-bold text-2xl text-primary dark:text-white placeholder-secondary focus:placeholder-transparent dark:bg-transparent dark:border-white focus:dark:border-tremor-secondary dark:placeholder-white focus:dark:placeholder-transparent"
    data-testid="pin-code-input"
    inputmode="numeric"
    maxlength="1"
    placeholder="â—‹"
    type="password"
    value=""
  />
</div>
`;
